src/3rd_party/FindBSON.cmake

```diff
@@ -0,0 +1,25 @@
+set(INCLUDE_PATH "${CMAKE_SOURCE_PREFIX}/include")
+set(LIB_PATH "${CMAKE_SOURCE_PREFIX}/lib")
+
+find_path(BSON_INCLUDE_DIRECTORY bson_object.h bson_array.h bson_util.h
+          PATHS "${INCLUDE_PATH}")
+
+find_library(BSON_LIB
+             NAMES bson
+             PATHS ${LIB_PATH})
+
+find_path(EMHASHMAP_INCLUDE_DIRECTORY emhashmap.h
+          PATHS ${INCLUDE_PATH}
+          PATH_SUFIXES emhashmap)
+
+find_library(EMHASHMAP_LIB
+             NAMES emhashmap
+             PATHS ${LIB_PATH})
+
+
+include(FindPackageHandleStandardArgs)
+
+find_package_handle_standard_args(BSON DEFAULT_MSG BSON_INCLUDE_DIRECTORY BSON_INCLUDE_DIRECTORY)
```

Comment:
`BSON_INCLUDE_DIRECTORY` is checked twice? This should be checking `BSON_INCLUDE_DIRECTORY`, `BSON_LIBS_DIRECTORY`, `EMHASHMAP_INCLUDE_DIRECTORY`, and `EMHASHMAP_LIBS_DIRECTORY`, correct?